packages/react-percy/src/cli.js

```diff
@@ -30,6 +30,22 @@ export function run(argv, rootDir) {
     return;
   }
 
+  if (!process.env.PERCY_TOKEN) {
+    process.stdout.write(
+      chalk.bold.red('PERCY_TOKEN') + chalk.red(' environment variable must be set.\n'),
+    );
+    process.on('exit', () => process.exit(1));
```

Comment:
We need to treat this as a warning and not an error, ie. we always need to exit from Percy with a 0 not a 1. If we exit with a 1, CI services will treat that as a failing step and abort the CI build and we want CI runs to run uninterrupted whether Percy is enabled or not.

This is how we do it in ember-percy (not super elegant):

```js
      if (environment.ci && !token) {
        console.warn(
          '[percy][WARNING] Percy is disabled, no PERCY_TOKEN environment variable found.')
      }
      if (environment.ci && !project) {
        console.warn(
          '[percy][WARNING] Percy is disabled, no PERCY_PROJECT environment variable found.')
      }
```

In some client libraries, we do error and fail if EITHER of the token or project is set, but not both. In that case it's more like a configuration error that should be fixed.

Reply:
@fotinakis One thing I'm wondering about is whether this approach (i.e. to not error) still makes sense if the snapshots aren't embedded in another test suite, but are their own process..  You think it does?

Reply:
Hmm yeah that behavior seems odd to me. I totally understand wanting a Percy outage not to cause users' CIs to fail, but if you're trying to run Percy with no token or project specified, that seems like it _should_ fail CI since the configuration is wrong. Otherwise it seems like it's failing silently and users who don't read their CI logs might expect Percy to be working correctly.

Reply:
@timhaines Maybe? It kind of depends on how  most people are going to set up their CI service to run Percy, like will it be `npm test && percy-snapshot` for all CI runs? Because if so, then yes we should do it. If we DON'T, then people have to invent their own bash conditions around `percy-snapshot` which we don't want to encourage.

I think it'd be fine to error if we supported `PERCY_ENABLE=0`, which is our documented way to "turn percy off" for a build.

Reply:
We do usually fail if only one of {project, token} is specified (a config error). But if they haven't set both of them, we usually warn instead.

I guess it's probably fine to error here (return 1) and try it in beta? As long as we supported PERCY_ENABLE=0

Reply:
Okay, I've left this as is for now, but happy to change it to whatever you guys like. I did update this to exit with 0 given a `PERCY_ENABLE=0` environment variable, though.